xlive/Blam/Engine/objects/object_identifier.cpp

```diff
@@ -1,37 +1,51 @@
 #include "stdafx.h"
 #include "object_identifier.h"
+#include "object_globals.h"
 
 void c_object_identifier::clear()
 {
 	this->m_object_type = (e_object_type)NONE;
 	this->m_source = (e_object_source)NONE;
 	this->m_origin_bsp_index = NONE;
 	this->m_unique_id = NONE;
+	return;
 }
 
 void c_object_identifier::clear_for_deletion()
 {
 	this->m_source = (e_object_source)NONE;
 	this->m_origin_bsp_index = NONE;
 	this->m_unique_id = NONE;
+	return;
 }
 
-short c_object_identifier::get_origin_bsp()
+void c_object_identifier::create_dynamic(e_object_type type)
+{
+	s_object_globals* object_globals = object_globals_get();
+	++object_globals->unique_id;
+	this->m_object_type = type;
+	this->m_source = _object_source_dynamic;
+	this->m_origin_bsp_index = NONE;
+	this->m_unique_id = object_globals->unique_id;
```

Comment:
Why not do object_globals->unique_id++; here instead of incrementing above

Reply:
Afaik pre-increment is faster assembly wise.

Reply:
https://stackoverflow.com/questions/38947790/efficiency-of-postincrement-v-s-preincrement-in-c

Reply:
Makes sense, something I would take into consideration though is if this will cause indexes to be skipped based on a potential conflict between being called with the native function and this one. However I have not looked at the native function and know if it is still called or not or if this is completely rerouted.

Reply:
changed to increment on assignment